packages/graph-retriever/src/graph_retriever/strategies/base.py

```diff
@@ -13,23 +13,36 @@
 class NodeTracker:
     """Helper class for tracking traversal progress."""
 
-    def __init__(self) -> None:
-        self.visited_ids: set[str] = set()
+    def __init__(self, select_k: int, max_depth: int| None) -> None:
+        self._select_k: int = select_k
+        self._max_depth: int | None = max_depth
+        self._visited_nodes: set[int] = set()
         self.to_traverse: dict[str, Node] = {}
         self.selected: dict[str, Node] = {}
 
+    @property
+    def remaining(self):
```

Comment:
Maybe `num_remaining` or something like that to emphasize it's a number? And maybe mentioning selected (`num_unselected`?)